sdk/storage/src/shares/share_client.cpp

```diff
@@ -217,4 +233,43 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         m_shareUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
   }
 
+  Azure::Core::Response<ListFilesAndDirectoriesSegmentedResult>
+  ShareClient::ListFilesAndDirectoriesSegmented(
```

Comment:
not align with other sdk

Reply:
Aligns with JS: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/src/ShareDirectoryClient.ts#L1179.

I don't think .NET is using a good name. I would imagine `Get` something returns the content, not the catalog.

Reply:
list is fine. I didn't mean the function name.

Reply:
the client